shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -79,16 +91,78 @@ private void init() {
       case surface:
         Log.d(TAG, "Internally creating a FlutterSurfaceView.");
         FlutterSurfaceView flutterSurfaceView = new FlutterSurfaceView(getContext());
+        renderSurface = flutterSurfaceView;
         addView(flutterSurfaceView);
         break;
       case texture:
         Log.d(TAG, "Internally creating a FlutterTextureView.");
         FlutterTextureView flutterTextureView = new FlutterTextureView(getContext());
+        renderSurface = flutterTextureView;
         addView(flutterTextureView);
         break;
     }
   }
 
+  /**
+   * Connects this {@code FlutterView} to the given {@link FlutterEngine}.
+   *
+   * Once invoked, the Flutter UI painted by the given {@link FlutterEngine} will be
+   * displayed by this {@code FlutterView}. Additionally, user touch events, accessibility
+   * events, keyboard events, and more will be forwarded from this {@code FlutterView}
+   * to the attached {@link FlutterEngine}.
+   *
+   * See {@link #detachFromFlutterEngine()} for information on how to detach from a
+   * {@link FlutterEngine}.
+   */
+  public void attachToFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+    if (isAttachedToFlutterEngine()) {
+      if (flutterEngine == this.flutterEngine) {
+        // We are already attached to this FlutterEngine
+        return;
+      }
+
+      // Detach from a previous FlutterEngine so we can attach to this new one.
+      detachFromFlutterEngine();
+    }
+
+    this.flutterEngine = flutterEngine;
+
+    // Instruct our FlutterRenderer that we are now its designated RenderSurface.
+    this.flutterEngine.getRenderer().attachToRenderSurface(renderSurface);
+  }
+
+  /**
+   * Disconnects this {@code FlutterView} from a previously attached {@link FlutterEngine}.
+   *
+   * Once invoked, the UI of this {@code FlutterView} will be cleared, and all touch events,
```

Comment:
> The UI ... will be cleared

I don't understand: Does this mean the screen will go black?

Reply:
Depends on which surface you're using.  I believe a SurfaceView will go black becuase it has a window.  I'm guessing that a TextureView would become transparent.  The actual implementation of that goal is tied to that commented out code you referenced earlier, so I can't actually test it right now to be sure.